File: sample/src/main/java/pl/droidsonroids/gif/sample/MainActivity.java
package pl.droidsonroids.gif.sample;

import android.os.Bundle;
import android.support.v4.app.FragmentActivity;
import android.support.v4.view.ViewPager;

import com.squareup.leakcanary.LeakCanary;
import com.squareup.leakcanary.RefWatcher;

/**
 * Main activity, hosts the pager
 *
 * @author koral--
 */
public class MainActivity extends FragmentActivity {

    private RefWatcher mRefWatcher;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        mRefWatcher = LeakCanary.install(getApplication());
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        ((ViewPager) findViewById(R.id.main_pager)).setAdapter(new MainPagerAdapter(this));
    }

    RefWatcher getRefWatcher() {
        return mRefWatcher;
    }

}

File: sample/src/main/java/pl/droidsonroids/gif/sample/MainActivity.kt
package pl.droidsonroids.gif.sample

import android.os.Bundle
import androidx.core.view.ViewCompat
import androidx.core.view.WindowInsetsCompat
import androidx.fragment.app.FragmentActivity
import androidx.viewpager2.widget.ViewPager2
import com.google.android.material.tabs.TabLayout
import com.google.android.material.tabs.TabLayoutMediator

/**
 * Main activity, hosts the pager

 * @author koral--
 */
class MainActivity : FragmentActivity() {

	override fun onCreate(savedInstanceState: Bundle?) {
		super.onCreate(savedInstanceState)
		setContentView(R.layout.activity_main)

        ViewCompat.setOnApplyWindowInsetsListener(window.decorView) { view, insets ->
            val systemBars = insets.getInsets(WindowInsetsCompat.Type.systemBars())
            view.setPadding(systemBars.left, systemBars.top, systemBars.right, systemBars.bottom)
            insets
        }

		val tabLayout = findViewById<TabLayout>(R.id.tabLayout)
		val pager = findViewById<ViewPager2>(R.id.mainViewPager)
		val mainPagerAdapter = MainPagerAdapter(this)

		pager.adapter = mainPagerAdapter
		TabLayoutMediator(tabLayout, pager) { tab, position ->
			tab.text = mainPagerAdapter.getPageTitle(position)
		}.attach()
	}
}
